src/gui/guiFormSpecMenu.cpp

```diff
@@ -4638,6 +4662,20 @@ bool GUIFormSpecMenu::OnEvent(const SEvent& event)
 			break;
 		}
 
+#ifdef HAVE_TOUCHSCREENGUI
+		if (touch == BET_RIGHT && m_selected_item) {
+			if (!s.isValid()) {
+				// Not a valid slot
+				if (!getAbsoluteClippingRect().isPointInside(m_pointer))
+					// Is outside the menu
+					drop_amount = 1;
+			} else {
+				// Over a valid slot
+				move_amount = 1;
```

Comment:
When you are dragging a full stack, you can't put an item back into the source slot with a second touch.

https://github.com/minetest/minetest/assets/82708541/73c39fcf-ae50-4fe9-a6c2-14bb77316c9c

Duplicating even more mouse event handling code fixes it:

```suggestion
				move_amount = 1;

				if (identical) {
					// Change the selected amount instead of moving
					if (move_amount >= m_selected_amount)
						m_selected_amount = 0;
					else
						m_selected_amount -= move_amount;
					move_amount = 0;
				}
```